### DISCUSSION TITLE
GPIO-Extender MCP23017 get to work with pipresents

### DISCUSSION THREAD

**Discussion by pete-b-tu** [KenT2/pipresents-gtk#7] [Q&A]
Hello Community and KenT2!
Thank you for your great work and the wonderful documentation on pipresents!
I am currently working on a project where I want to use pipresents with a lot of Sensors and LED's so the given GPIO's form the pi won't be enough to drive the presentation. I found the [MCP23017](https://www.adafruit.com/product/5346) and ordered a few and wanted them to expand the GPIO's to a higher number. The MCP23017 is driven by the I2c protocol and delivers the possibilities to get 128 GPIO's in addition. For the MCP23017 exists a [python library](https://learn.adafruit.com/adafruit-mcp23017-i2c-gpio-expander) to get it work with the pi. So now I am wondering if you could give me some hints where to establish or connect the code to use it with the pi and pi presents and connect the new GPIO's to the gpiozero.cfg (or another?). Do you think this will be possible? I have a bit knowledge about programming with python and raspipi so I am not entirely new to programming and maybe when you tell me where to add the program code i may realize it by myself :).

I am hoping for your feedback!
Kind regards, Peter

**Comment by KenT2** [comment]
You need to write an I/O plugin. This will be a python program which will probably act as an interface between the python library and Pi Presents. There is already an I/O plugin pp_i2cdriver.py which uses other I2C based libraries. \the plugin would have its own configuration file, mcp2017.cfg maybe, which could be similar to gpiozero.cfg.

Ypu may need to modify the Adafruit python library as it uses Circuit Python

I would be interested to add any plugin you produce to the Pi presents release.

If you are using pp-gtk note that the video player is still unreliable.
